pos-print/src/main/java/com/clouway/pos/print/adapter/http/PrintService.kt

```diff
@@ -25,22 +26,33 @@ class PrintService @Inject constructor(private var factory: PrinterFactory) {
     try {
       val dto = request.read(ReceiptDTO::class.java).`as`(GsonTransport::class.java)
       val printer = factory.getPrinter(dto.sourceIp)
-      when {
-        dto.fiscal
-        -> response = printer.printFiscalReceipt(dto.receipt)
-        else
-        -> response = printer.printReceipt(dto.receipt)
+      try {
+        when {
+          dto.fiscal
+          -> response = printer.printFiscalReceipt(dto.receipt)
+          else
+          -> response = printer.printReceipt(dto.receipt)
+        }
+      } catch (e: RequestTimeoutException) {
+        e.printStackTrace()
+        return Reply.with(ErrorResponse("Printer request timeout.")).`as`(GsonTransport::class.java).badRequest()
```

Comment:
> 417 EXPECTATION FAILED
The expectation given in the request's Expect header field1 could not be met by at least one of the inbound servers.

Doesn't sound reasonable to me, as it's used for indication of missing headers.